canopeum_frontend/src/pages/Register.tsx

```diff
@@ -9,152 +11,41 @@ import useApiClient from '@hooks/ApiClientHook'
   const [codeInvalid, setCodeInvalid] = useState(false)
   const [codeExpired, setCodeExpired] = useState(false)
 
   const [userInvitation, setUserInvitation] = useState<UserInvitation>()
 
-  const fetchUserInvitation = useCallback(async (code: string) => {
-    try {
-      const userInvitationResponse = await getApiClient().userInvitationClient.detail(code)
-      if (userInvitationResponse.expiresAt <= new Date()) {
-        setCodeExpired(true)
-        setCodeInvalid(false)
-        setUserInvitation(undefined)
-
-        return
-      }
-
-      setCodeExpired(false)
-      setCodeInvalid(false)
-      setUserInvitation(userInvitationResponse)
-      setEmail(userInvitationResponse.email)
-    } catch {
-      setCodeInvalid(true)
-      setCodeExpired(false)
-      setUserInvitation(undefined)
-    }
-  }, [getApiClient])
-
-  useEffect(() => {
-    const code = searchParams.get('code')
-    if (!code) return
-
-    void fetchUserInvitation(code)
-  }, [searchParams, fetchUserInvitation])
-
-  const validateUsername = () => {
-    if (!username) {
-      setUsernameError('required')
-
-      return false
-    }
-
-    setUsernameError(undefined)
-
-    return true
-  }
-
-  const validateEmail = () => {
-    if (!email) {
-      setEmailError('required')
-
-      return false
-    }
-
-    if (!isValidEmail(email)) {
-      setEmailError('email')
-
-      return false
-    }
-
-    setEmailError(undefined)
-
-    return true
-  }
-
-  const validatePassword = () => {
-    if (!password) {
-      setPasswordError('required')
-
-      return false
-    }
-
-    if (!isValidPassword(password)) {
-      setPasswordError('password')
-
-      return false
-    }
-
-    setPasswordError(undefined)
-
-    return true
-  }
-
-  const validatePasswordConfirmation = () => {
-    if (!passwordConfirmation) {
-      setPasswordConfirmationError('required')
-
-      return false
-    }
-
-    if (!mustMatch(password, passwordConfirmation)) {
-      setPasswordConfirmationError('mustMatch')
-
-      return false
-    }
-
-    setPasswordConfirmationError(undefined)
-
-    return true
-  }
-
-  const validateForm = () => {
-    // Do not return directly the method calls;
-    // we need each of them to be called before returning the result
-    const usernameValid = validateUsername()
-    const emailValid = validateEmail()
-    const passwordValid = validatePassword()
-    const passwordConfirmationValid = validatePasswordConfirmation()
-
-    return usernameValid
-      && emailValid
-      && passwordValid
-      && passwordConfirmationValid
-  }
-
-  const onCreateAccountClick = async () => {
-    const isFormValid = validateForm()
-    if (!isFormValid) return
-
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, touchedFields },
+    setValue,
+  } = useForm<RegisterInputs>({ mode: 'onTouched' })
+  const onSubmit: SubmitHandler<RegisterInputs> = async data => {
```

Comment:
Nitpick, but could the variable be renamed `registerInputs` or `formData` ? And maybe the type to `RegisterFormInputs` ? Just to help with readability and avoid confusion. Thoughts?
```suggestion
  const onSubmit: SubmitHandler<RegisterFormInputs> = async formData=> {
```